pkg/registry/client.go

```diff
@@ -105,10 +105,11 @@ func NewClient(options ...ClientOption) (*Client, error) {
 			if err != nil {
 				return nil, errors.New("unable to retrieve credentials")
```

Comment:
Not directly related to this issue, but I am wondering if swallowing the error over doing `fmt.Errorf("unable to retrieve credentials: %w, err)` is the best thing to do?

Reply:
We shouldn't swallow the error.

Reply:
I'll fill a separate PR to address this to allow you to move on with this quickly.